src/elementary/src/toEndfReactionNumber.hpp

```diff
@@ -19,22 +21,72 @@ namespace elementary {
    */
   int toEndfReactionNumber( const ReactionID& identifier ) {
 
-    // treat special cases
+    if ( not identifier.isSpecial() ) {
 
-    // other cases
+      // get the incident and outgoing particle information
+      const auto incident = identifier.incident();
+      const auto outgoing = identifier.outgoing();
+      const auto incidentResidual = incident.residual();
+      const auto outgoingParticles = outgoing.particles();
+      const auto outgoingResidual = outgoing.residual();
 
-    // get the incident and outgoing particle information
-    const auto incident = identifier.incident();
-    const auto outgoing = identifier.outgoing();
-//    const auto incidentParticle = incident.particle();
-    const auto incidentResidual = incident.residual();
-    const auto outgoingParticles = outgoing.particles();
-    const auto outgoingResidual = outgoing.residual();
+      // elastic scattering is a special case
+      if ( incidentResidual == outgoingResidual ) {
 
-    // elastic scattering
-    if ( incidentResidual == outgoingResidual ) {
+        return 2;
+      }
+      else {
 
-      return 2;
+        std::array< unsigned int, 6 > counts = {{ 0, 0, 0, 0, 0, 0 }};

```

Comment:
I don't follow the logic here. What exactly are you trying to accomplish?

Reply:
I'm counting the number of each particle in the list of particles, in order, to generate a string like "2n2a" so that I can just look up the corresponding reaction in the registered reactions.